server/server/src/cold_chain/temperature_breach.rs

```diff
@@ -0,0 +1,188 @@
+use std::convert::TryInto;
+
+use actix_web::{
+    http::header,
+    web::{self, Data},
+    HttpRequest, HttpResponse,
+};
+use anyhow::Context;
+use chrono::NaiveDateTime;
+use log::error;
+use mime_guess::mime;
+use repository::{RepositoryError, TemperatureBreachRowType};
+use service::{
+    auth_data::AuthData,
+    service_provider::{ServiceContext, ServiceProvider},
+    temperature_breach::{insert::InsertTemperatureBreach, update::UpdateTemperatureBreach},
+    SingleRecordError,
+};
+use util::constants::SYSTEM_USER_ID;
+
+use super::validate_request;
+
+#[derive(serde::Deserialize, Debug, Clone)]
+pub struct TemperatureBreach {
+    id: String,
+    acknowledged: bool,
+    #[serde(rename = "endTimestamp")]
+    end_unix_timestamp: Option<i64>,
+    #[serde(rename = "sensorId")]
+    sensor_id: String,
+    #[serde(rename = "startTimestamp")]
+    start_unix_timestamp: i64,
```

Comment:
could still use rename_all at structure level and separate rename for this one

Reply:
yeah, there's only two fields that the rename_all applies to so it didn't seem worth it 🤷